### ISSUE TITLE
Headers: no catchable exception

### ISSUE THREAD

**Issue by wujku** [ddeboer/imap#246]
When date is incorrect, I can not get any of message header.
https://github.com/ddeboer/imap/blob/0788d5d444001e252b3dcc8fa2d4546702a2ef5b/src/Message/Headers.php#L65

I suggest to parse `date` header when user use `getDate()` method or return false instead of InvalidDateHeaderException.

```
try {
    $message->getId();
} catch (InvalidDateHeaderException $e) {
    ???
}
```

**Comment by Slamdunk** [comment]
I'll look at this, hoping to push a fix without a BC break.

In the meanwhile, can you post please the Date header that produced the exception?

**Comment by Slamdunk** [comment]
Lazy loading the parameters is a good idea, unfortunately `Headers` extends `ArrayIterator` and lazy loading an item involves too much code overwrite, because a user can access to a header through a lot of methods (`current`, `getArrayCopy`, `next`, `offsetGet`, `get`, `seek`, `serialize` etc).

I'm going to tag this for v2.0, but I'm going to release a major version not before 6 months from now.

**Comment by wujku** [comment]
This is a raw email which cause this error. 
I fetch messages from imap account, which I used to phpunit tests.

```
Date: Thu, 05 Oct 2017 13:14:32 +0000
From: from@here.com
To: to@there.com
Subject: Message 2
MIME-Version: 1.0
Content-Type: text/plain
Content-Transfer-Encoding: 8bit

Message 259d6303839b75
```

**Comment by Slamdunk** [comment]
`Thu, 05 Oct 2017 13:14:32 +0000` doesn't raise an Exception, it is correctly parsed to `2017-10-05T13:14:32+0000`

**Comment by wujku** [comment]
@Slamdunk Sorry, this mail has below content:

```
Subject: test
MIME-Version: 1.0
Content-Type: text/plain
Date: Fri!
From: from@there.com
To: to@here.com

Hi!
```

**Comment by wujku** [comment]
@Slamdunk Do you have idea how catch this exception and have ability to access other header variables?

**Comment by Slamdunk** [comment]
At the time being, no

**Comment by Slamdunk** [comment]
Implemented in #250
